parsons/quickbooks/quickbookstime.py

```diff
@@ -0,0 +1,732 @@
+
+            [QuickBooksTime API Documentation](https://tsheetsteam.github.io/api_docs/#introduction)
+    """
+
+    def __init__(self, token=None):
+        self.token = check_env.check("QB_AUTH_TOKEN", token)
+        self.headers = {"Authorization": "Bearer " + self.token}
+        self.client = APIConnector(QB_URI, headers=self.headers)
+
+    # Helper functions
+
+    def qb_get_request(self, end_point: str, querystring=None):
+        """
+        This function handles the pagination of the request
+        """
+
+        # If no querystring is provided, initialize it as an empty dictionary
+        if querystring is None:
+            querystring = {}
+        output_list = []  # This list will hold the results
+
+        page = 0  # Start from the first page
+        more = True  # This flag indicates if there are more pages to fetch
+        while more:
+            # After every 10 pages, log the progress
+            if page % 10 == 0:
+                logger.info(f"Retrieved {len(output_list)} records from {end_point} endpoint.")
+                logger.info(f"Currently on page {page}.")
+
+            # Add the current page to the querystring
+            querystring = {**querystring, **{"page": page}}
+
+            # Send the GET request
+            response = self.client.get_request(end_point, params=querystring)
+
+            # Extract the key of the results
+            endpoint_key = list(response["results"].keys())[0]
+            # Extract the records from the results
+            temp_list = [
+                response["results"][endpoint_key][record]
+                for record in response["results"][endpoint_key]
+            ]
+
+            # If the response indicates there are more pages,
+            # update the flag and increment the page number
+            if "more" in response:
+                more = response["more"]
+                page += 1
+
+            # Add the records from the current page to the output list
+            output_list += temp_list
+
+            # If there are no more pages, break the loop
+            if "more" not in response or response["more"] is False:
+                break
+
+        # Log the total number of records retrieved
+        logger.info(f"Retrieved {len(output_list)} records from {end_point} endpoint.")
+        # Return the results as a Table
+        return Table(output_list)
+
+    def get_groups(
+        self,
+        ids=None,
+        active=None,
+        manager_ids=None,
+        name=None,
+        modified_before=None,
+        modified_since=None,
+        supplemental_data=None,
+        per_page=None,
+        page=None,
+    ):
+        """
+            This function allows you to call the /groups endpoint of the QuickBooksTime Time API.
+            All Args are optional.
+
+            `Args:`
+                ids: Int
```

Comment:
It seems like there might be an indentation issue here? Can you fix it so all args are at the same level of indentation?